tests/functional/spec/dates.spec.js

```diff
@@ -0,0 +1,129 @@
+import chai from 'chai'
```

Comment:
Do we need to exercise all of this logic as selenium tests, can the validation logic not be exercised as unit tests?